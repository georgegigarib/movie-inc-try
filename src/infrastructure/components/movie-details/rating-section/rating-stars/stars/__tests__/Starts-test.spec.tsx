import { render, fireEvent } from '@testing-library/react-native';
import StarRow from '@/src/infrastructure/components/movie-details/rating-section/rating-stars/stars/Stars';
import React from 'react';

describe('StarRow', () => {
  test('renders correctly and handles press events', () => {
    const handlePress = jest.fn();
    const { getByTestId, toJSON } = render(
      <StarRow
        start={1}
        count={5}
        rating={3}
        submitted={false}
        onPress={handlePress}
      />
    );

    for (let i = 1; i <= 5; i++) {
      const star = getByTestId(`star-row-touchable-${i}`);
      expect(star).toBeTruthy();
    }

    fireEvent.press(getByTestId('star-row-touchable-2'));
    expect(handlePress).toHaveBeenCalledWith(2);
    expect(toJSON()).toMatchSnapshot();
  });
});
