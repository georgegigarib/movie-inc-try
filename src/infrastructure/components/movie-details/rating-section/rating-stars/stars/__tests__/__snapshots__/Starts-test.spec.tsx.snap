// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRow renders correctly and handles press events 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="star-row-touchable-1"
  >
    <View
      collapsable={false}
      style={
        {
          "marginHorizontal": 4,
        }
      }
      testID="star-row-animated-view-1"
    >
      <Text
        style={
          {
            "color": "#FFD700",
            "fontSize": 24,
          }
        }
      >
        star
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="star-row-touchable-2"
  >
    <View
      collapsable={false}
      style={
        {
          "marginHorizontal": 4,
        }
      }
      testID="star-row-animated-view-2"
    >
      <Text
        style={
          {
            "color": "#FFD700",
            "fontSize": 24,
          }
        }
      >
        star
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="star-row-touchable-3"
  >
    <View
      collapsable={false}
      style={
        {
          "marginHorizontal": 4,
        }
      }
      testID="star-row-animated-view-3"
    >
      <Text
        style={
          {
            "color": "#FFD700",
            "fontSize": 24,
          }
        }
      >
        star-o
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="star-row-touchable-4"
  >
    <View
      collapsable={false}
      style={
        {
          "marginHorizontal": 4,
        }
      }
      testID="star-row-animated-view-4"
    >
      <Text
        style={
          {
            "color": "#FFD700",
            "fontSize": 24,
          }
        }
      >
        star-o
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="star-row-touchable-5"
  >
    <View
      collapsable={false}
      style={
        {
          "marginHorizontal": 4,
        }
      }
      testID="star-row-animated-view-5"
    >
      <Text
        style={
          {
            "color": "#FFD700",
            "fontSize": 24,
          }
        }
      >
        star-o
      </Text>
    </View>
  </View>
</View>
`;
